rfqmsg/custom_channel_data.go

```diff
@@ -28,19 +28,19 @@ type JsonAssetUtxo struct {
 	DecimalDisplay uint8            `json:"decimal_display"`
 }
 
-// JsonAssetChanInfo is a struct that represents the channel information of a
-// single asset within a channel.
-type JsonAssetChanInfo struct {
-	AssetInfo     JsonAssetUtxo `json:"asset_utxo"`
-	Capacity      uint64        `json:"capacity"`
-	LocalBalance  uint64        `json:"local_balance"`
-	RemoteBalance uint64        `json:"remote_balance"`
-}
-
 // JsonAssetChannel is a struct that represents the channel information of all
 // assets within a channel.
 type JsonAssetChannel struct {
-	Assets []JsonAssetChanInfo `json:"assets"`
+	FundingAssets       []JsonAssetUtxo    `json:"funding_assets"`
+	LocalAssets         []JsonAssetTranche `json:"local_assets"`
+	RemoteAssets        []JsonAssetTranche `json:"remote_assets"`
+	OutgoingHtlcs       []JsonAssetTranche `json:"outgoing_htlcs"`
+	IncomingHtlcs       []JsonAssetTranche `json:"incoming_htlcs"`
+	Capacity            uint64             `json:"capacity"`
+	LocalBalance        uint64             `json:"local_balance"`
```

Comment:
Will there be one of these new structs for each future asset group? 

Also when it comes to balances, will we show a sum of the total grouped assets, but then individual balances for each asset ID included?

Reply:
There will only be a single `JsonAssetChannel` per channel. But that will now more clearly show the difference between the funding blob (e.g. the assets that were committed to the channel at funding time --> `FundingAssets`, for example `25k units of asset A, 30k units of asset B`) and the current commitment blob (e.g. the current distribution of assets --> `Local/RemoteBalance/Outgoing/IncomingHtlcs`, for example `LocalBalance: 10k units of asset A, RemoteBalance: 15k units of asset A, 25k units of asset B, OutgoingHtlc: 5k units of asset B`).